dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathEntryTest.java

```diff
@@ -78,6 +86,38 @@ public void testToStringArtifact() {
     assertEquals("com.google:foo:0.0.1", entry.toString());
   }
 
+  @Test
+  public void testListFileNames() throws IOException, ArtifactResolutionException {
+    
+    // copy into the local repository so we can read the jar file
+    Artifact artifact = resolveArtifact("com.google.truth.extensions:truth-java8-extension:1.0.1");
+    
+    ClassPathEntry entry = new ClassPathEntry(artifact);
+    ImmutableSet<String> classFileNames = entry.listClassFileNames();
+    Truth.assertThat(classFileNames).containsExactly(
+        "com.google.common.truth.IntStreamSubject",
+        "com.google.common.truth.LongStreamSubject",
+        "com.google.common.truth.OptionalDoubleSubject",
+        "com.google.common.truth.OptionalSubject",
+        "com.google.common.truth.OptionalIntSubject",
+        "com.google.common.truth.OptionalLongSubject",
+        "com.google.common.truth.PathSubject",
+        "com.google.common.truth.Truth8",
+        "com.google.common.truth.StreamSubject");
```

Comment:
You may want to add some inner classes.

Reply:
yes, looking at that now.

Reply:
Also verify there's no such class that has package name `META-INF`?